api/src/main/java/edu/wpi/first/shuffleboard/api/components/ActionList.java

```diff
@@ -0,0 +1,90 @@
+package edu.wpi.first.shuffleboard.api.components;
+
+import edu.wpi.first.shuffleboard.api.util.FxUtils;
+import edu.wpi.first.shuffleboard.api.util.TypeUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import javafx.scene.Node;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuItem;
+
+public class ActionList {
+  private final String name;
+  private final List<Supplier<MenuItem>> actions;
+
+  private ActionList(String name) {
+    this.name = name;
+    this.actions = new ArrayList<>();
+  }
+
+  public ActionList addAction(String name, Runnable r) {
+    actions.add(() -> FxUtils.menuItem(name, _e -> r.run()));
```

Comment:
I typically use a double underscore `__` for unused lambda parameters. Easier to read IMO